src/protagonist/DLCS.Repository/Spaces/SpaceRepository.cs

```diff
@@ -207,4 +208,37 @@ private async Task<int> GetIdForNewSpace(int requestCustomer)
         var retrievedSpace = await GetSpaceInternal(customerId, spaceId, cancellationToken);
         return retrievedSpace;
     }
+
+    public async Task<(DeleteResult, string)> DeleteSpace(int customerId, int spaceId, CancellationToken cancellationToken)
+    {
+        var deleteResult = DeleteResult.NotFound;
+        var message = string.Empty;
+        
+        var space = await dlcsContext.Spaces.AsNoTracking().SingleOrDefaultAsync(s =>
+            s.Customer == customerId && s.Id == spaceId, cancellationToken: cancellationToken);
+
+        if (space != null)
+        {
+            var images = await dlcsContext.Images.AsNoTracking().Where(i => 
+                i.Customer == customerId && i.Space == spaceId).CountAsync(cancellationToken);
+            
+            if (images == 0)
+            {
+                dlcsContext.Spaces.Attach(space);
+                dlcsContext.Spaces.Remove(space);

```

Comment:
To avoid needing to `.Attach()` and then `.Remove()` the space prior to deleting. Alter the query fetching space to remove `.AsNoTracking()`.

```cs
var space = await dlcsContext.Spaces.SingleOrDefaultAsync(s =>
            s.Customer == customerId && s.Id == spaceId, cancellationToken: cancellationToken);
```